### PR TITLE
Runtime change required for bridge to work

### PR DESCRIPTION
**Pull request by ParthDesai**
# Description
This PR contains subset of changes from https://github.com/moondance-labs/tanssi/pull/733 related to runtime changes. Namely following: 
1. Renaming `MMRLeaf` pallet to `BeefyMMRLeaf` so that snowbridge scripts would work.
2. Adding `FrameMetadataHashExtension` so that we are able to parse transactions sent by snowbridge.

# Note
Regarding `FrameMetadataHashExtension`, the capability of generating the metadata hash at compile time is not added in dancelight runtime. So, transactions with the `mode` set to `enabled` for this extension will fail.

### CODE REVIEW

**Review by girazoki — APPROVED** [review]
looks good @ParthDesai good job!

**Comment by ParthDesai** [comment]
> looks good @ParthDesai good job!

@girazoki Reverted spec version increment.

File: solo-chains/runtime/dancelight/src/tests/migrations_test.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+use crate::{BeefyMmrLeaf, PalletInfo, Runtime};
+use beefy_primitives::mmr::BeefyAuthoritySet;
+use frame_support::storage::unhashed;
+use frame_support::traits::PalletInfo as _;
+use pallet_migrations::Migration;
+use tanssi_runtime_common::migrations::MigrateMMRLeafPallet;
+use xcm::v3::Weight;
+
+#[test]
+fn test_migration_config_full_rotation_period() {
```
Comment (tmpolaczyk) [inline solo-chains/runtime/dancelight/src/tests/migrations_test.rs:27]: test name

**Review by tmpolaczyk — APPROVED** [review]

File: solo-chains/runtime/dancelight/src/lib.rs [hunk 2]
```diff
@@ -631,6 +631,7 @@ where
             frame_system::CheckNonce::<Runtime>::from(nonce),
             frame_system::CheckWeight::<Runtime>::new(),
             pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+            frame_metadata_hash_extension::CheckMetadataHash::<Runtime>::new(false),
```
[the remark this answers is not in the document]
Comment (ParthDesai) [reply solo-chains/runtime/dancelight/src/lib.rs:634]: yes. It means the extension is disabled.

**Review by nanocryk — APPROVED** [review]

File: solo-chains/runtime/dancelight/src/tests/migrations_test.rs (hunk 1 above)
Comment (ParthDesai) [reply solo-chains/runtime/dancelight/src/tests/migrations_test.rs:27]: *fixed*

**Pull request merged**